### ISSUE TITLE
ACSet Homomorphism Serialization

### ISSUE THREAD

**Issue by jpfairbanks** [AlgebraicJulia/Catlab.jl#678]
We need to extend the `write_json_acset` function to write CSet homomorphisms. @kris-brown, have you done something similar for any of your projects?

**Comment by kris-brown** [comment]
I have not. No special insights - I’d just do “dom” “codom” and “components” keys. If we wanted to do diagram to json then this would be a special case for walking arrow shape.

**Comment by jpfairbanks** [comment]
Yeah, the diagram of ACSet serialization would make sense. I guess we would store the FreeDiagram ACSet to store the shape of the diagram and then each of the objects in the diagram in a list and then each of the arrows as just a named tuple of the underlying components. We should probably do that rather than a special case for a single arrow. 

Is there anything to do about redundancy? It seems like for many cases the Dom, Codom, components encoding will be wasteful.
